modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/add_to_cart/AddToCartWrapper.js

```diff
@@ -0,0 +1,247 @@
+	const CartResource = ServiceProvider.DeliveryCartAPI('v1');
+	const [accountId, setAccountId] = useState(props.accountId);
+	const [orderQuantity, setOrderQuantity] = useState(props.orderQuantity);
+	const [orderId, setOrderId] = useState(props.orderId);
+	const [multipleQuantity, setMultipleQuantity] = useState(
+		props.settings.multipleQuantity
+	);
+	const [selectedQuantity, setSelectedQuantity] = useState();
+	const [disabled, setDisabled] = useState();
+	const [skuId, setSkuId] = useState(props.skuId);
+
+	useEffect(() => {
+		setMultipleQuantity(props.settings.multipleQuantity);
+		if (props.settings.allowedQuantity?.length > 0) {
+			setOrderQuantity(props.settings.allowedQuantity);
+			setMultipleQuantity(1);
+		}
+		else {
+			const quantities = generateOptions(props.settings);
+			setOrderQuantity(quantities);
+		}
+		if (props.settings.multipleQuantity === undefined) {
+			setMultipleQuantity(1);
+		}
+		if (props.preSelectedQuantity) {
+			setSelectedQuantity(props.preSelectedQuantity);
+		}
+	}, [props.settings, props.orderQuantity, props.preSelectedQuantity]);
+
+	useEffect(() => {
+		if (props.skuId !== skuId) {
+			setSkuId(props.skuId);
+		}
+		function handleAccountUpdated({id}) {
+			setAccountId(id);
+		}
+		if (props.accountId !== accountId) {
+			setAccountId(props.accountId);
+		}
+		if (accountId === undefined || skuId === undefined) {
+			setDisabled(true);
+		}
+		else {
+			setDisabled(false);
+		}
+		Liferay.on(CURRENT_ACCOUNT_UPDATED, handleAccountUpdated);
+
+		return () => {
+			Liferay.detach(CURRENT_ACCOUNT_UPDATED, handleAccountUpdated);
+		};
+	}, [accountId, props.accountId, props.skuId, skuId]);
+
+	useEffect(() => {
+		if (!accountId || !props.skuId || !orderId) {
+			setDisabled(true);
+			showNotification('no-account-selected', 'danger', true, 500);
+		}
+		else {
+			setDisabled(false);
+		}
+	}, [accountId, props.skuId, orderId]);
+
+	const updateQuantity = (who, value) => {
+		if (!value && !selectedQuantity) {
+			value = 1;
+			setSelectedQuantity(1);
+		}
+		else if (!value && selectedQuantity) {
+			value = selectedQuantity;
+		}
+		else {
+			setSelectedQuantity(value);
+		}
+
+		const quantity = multipleQuantity * value;
+		const itemPayload = {
+			quantity,
+			skuId,
+		};
+
+		if (
+			who === 'button' ||
+			(who === 'select' && props.disableAddToCartButton)
+		) {
+			if (!orderId) {
+				CartResource.createCartByChannelId(props.channelId, {
+					accountId: props.accountId,
+					cartItems: [itemPayload],
+					currencyCode: props.currencyCode,
+				}).then((data) => {
+					if (orderId !== data.id) {
+						setOrderId(data.id);
+					}
+				});
+			}
+			else {
+				CartResource.createItemByCartId(orderId, {
+					...itemPayload,
+					productId: props.productId,
+				});
+			}
+		}
+	};
+
+	if (!props.disableAddToCartButton || !props.disableQuantitySelector) {
+		return (
+			<ClayIconSpriteContext.Provider value={props.spritemap}>
+				{props.settings.allowedQuantity?.length > 0 && (
```

Comment:
@gianmarcobrunialti @garofalog those settings should go in the quantity selector as are shared as well in other consumer, like the mi cart.  is this right?

Reply:
these settings goes there cause QS it's strictly related to AddToCartWrapper. QS should be used alone but we might use ClaySelecy instead of this high level component